std/algorithm/iteration.d

```diff
@@ -2695,12 +2740,110 @@ version (none) // this example requires support for non-equivalence relations
     }
 }
 
-// https://issues.dlang.org/show_bug.cgi?id=13595
-version (none) // This requires support for non-equivalence relations
+
+
+// https://issues.dlang.org/show_bug.cgi?id=13805
 @system unittest
+{
+    [""].map!((s) => s).chunkBy!((x, y) => true);
+}
+
+/**
+Like $(REF chunkBy), but does not require reflexivity, symmetricity nor
+transitivity of the binary predicate. Unlike `çhunkBy`, requires the
+source to be a forward range.
+With binary predicates, the element closer to front is always passed
+first to the predicate.
+When called with unary predicate, the comparisons are still assumed
+to be reflexive, symmetric and transitive. However, the source range is
+nonetheless required to be a forward range.
```

Comment:
Is this limitation really necessary?

Reply:
Which one - that the source range needs to be a forward range, or that we require the potential `==` overload of the element type to be reflexive, symmetric and transitive?

Reply:
That the source needs to be a forward range. Unless I'm missing something, this function has the same behavior as regular `chunkBy` in the unary-predicate case, so it's strange for it to have different requirements.

Reply:
The logic goes like this: If the user has used `chunkByAny` instead of `chunkBy`, she probably wants to make sure that if the predicate is changed to a binary one, it's reflexivity, symmetricity and transitivity are not assumed. So we check that the code would compile even if we could not assume the usual things about the predicate.

That the equivalency is still assumed for unary predicates is just a kind of optimization - we assume the user is not ridiculous enough to define a `==` overload that would not satisfy `chunkBy` requirements.

Reply:
If the idea is that `chunkByAny` (or `splitBy`) should only be used when the programmer intends to use a binary predicate (either now or in the future), then maybe the right thing to do is to *only* accept binary predicates, and leave the unary predicate case to regular `chunkBy`.

Reply:
I might otherwise argue with this, but as we're taking some distance from `chunkBy` anyway with the refactoring to `splitBy`, I can just as well do that.